addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -3202,31 +3202,49 @@ var JournalDashboardGraph = AbstractField.extend({
      * @override
      * @private
      */
-    _render: function () {
+    _render: async function () {
         if (this._isInDOM) {
-            return this._renderInDOM();
+            return await this._renderInDOM();
         }
-        return Promise.resolve();
+        return;
     },
     /**
      * Render the widget. This function assumes that it is attached to the DOM.
      *
      * @private
      */
-    _renderInDOM: function () {
-        this.$el.empty();
+    _renderInDOM: async function () {
         var config, cssClass;
+        var self = this;
         if (this.graph_type === 'line') {
-            config = this._getLineChartConfig();
             cssClass = 'o_graph_linechart';
         } else if (this.graph_type === 'bar') {
-            config = this._getBarChartConfig();
             cssClass = 'o_graph_barchart';
         }
         this.$canvas = $('<canvas/>');
         this.$el.addClass(cssClass);
         this.$el.empty();
         this.$el.append(this.$canvas);
+
+        if (!this.data) {
+            this.data = [await this._rpc({
```

Comment:
fetching data in ``_renderInDOM`` doesn't look as a good idea.

Anyway, are you sure in if-condition ? What if data is already fetched, but contains False?

Reply:
Not a good idea because of MVC pattern, performances, or because of something else?

And no, the if-condition should be changed indeed

Reply:
> Not a good idea because of MVC pattern, performances, or because of something else?

MVC